cmd/soroban-cli/src/commands/tx/sign.rs

```diff
@@ -0,0 +1,97 @@
+use std::io;
+
+// use crossterm::{
+//     event::{self, Event, KeyCode, KeyEvent, KeyModifiers},
+//     execute,
+//     terminal::{self, EnterAlternateScreen, LeaveAlternateScreen},
+// };
+use soroban_sdk::xdr::{self, Limits, TransactionEnvelope, WriteXdr};
+
+use crate::signer::{self, InMemory, Stellar};
+
+#[derive(thiserror::Error, Debug)]
+pub enum Error {
+    #[error(transparent)]
+    XdrArgs(#[from] super::xdr::Error),
+    #[error(transparent)]
+    Signer(#[from] signer::Error),
+    #[error(transparent)]
+    Config(#[from] super::super::config::Error),
+    #[error(transparent)]
+    StellarStrkey(#[from] stellar_strkey::DecodeError),
+    #[error(transparent)]
+    Xdr(#[from] xdr::Error),
+    #[error(transparent)]
+    Io(#[from] io::Error),
+    #[error("User cancelled signing, perhaps need to add -y")]
+    UserCancelledSigning,
+}
+
+#[derive(Debug, clap::Parser, Clone)]
+#[group(skip)]
+pub struct Cmd {
+    /// Confirm that a signature can be signed by the given keypair automatically.
+    #[arg(long, short = 'y')]
+    yes: bool,
+    #[clap(flatten)]
+    pub xdr_args: super::xdr::Args,
+    #[clap(flatten)]
+    pub config: super::super::config::Args,
+}
+
+impl Cmd {
+    #[allow(clippy::unused_async)]
+    pub async fn run(&self) -> Result<(), Error> {
+        let envelope = self.sign()?;
+        println!("{}", envelope.to_xdr_base64(Limits::none())?.trim());
```

Comment:
What is the `.trim()` needed for?

Reply:
Had issues with piping to a send command. But perhaps it should be on the input.